stash/client_repositories_user.go

```diff
@@ -128,7 +128,7 @@ func (c *UserRepositoriesClient) Create(ctx context.Context,
 		return nil, err
 	}
 
-	apiObj, err := createRepository(ctx, c.client, addTilde(ref.UserLogin), ref, req, opts...)
+	apiObj, err := createRepository(ctx, c.client, addTilde(c.client.username), ref, req, opts...)
```

Comment:
So `func NewStashClient(username, token string, optFns ...gitprovider.ClientOption) (*ProviderClient, error)` should be changed to loose the `username`.

`func WithAuth(username string, token string) ClientOptionsFunc` as well.

Reply:
I don't understand. The username is needed to properly authenticate.

Reply:
I'm going to merge this now for the sake of fixing the original issue but am happy to figure out what do to about that username var.